test/test_torch.py

```diff
@@ -3276,6 +3276,24 @@ def warn_fn():
             self.assertEqual(frameinfo.lineno - 6, warning.lineno)
             self.assertEqual(len(w), 1)
 
+    @onlyCPU
+    def test_warn_always_caught(self, device):
+        # Check that we can catch a TORCH_WARN_ONCE warning twice
+        # since mustWarnRegex uses set_warn_always(True) which changes
+        # TORCH_WARN_ONCE to TORCH_WARN
+        a = np.arange(10)
+        a.flags.writeable = False
+        with self.mustWarnRegex(UserWarning, '.*non-writeable.*'):
+            torch.from_numpy(a)
+
+        with self.mustWarnRegex(UserWarning, '.*non-writeable.*'):
+            torch.from_numpy(a)
+
+        with self.assertRaisesRegex(AssertionError, '.*too many.*non-writeable.*'):
```

Comment:
Would you elaborate on what's going on here?

Reply:
I will add comments

Reply:
The idea is to check that 
- two consecutive warnings will both be caught, and that 
- the decorator will raise if more than one warning is raised in a test